docs/eBpfExtensions.md

```diff
@@ -143,6 +147,14 @@ This structure is used to specify the address at which the various helper functi
 * `helper_function_count`: Number of helper functions implemented by the extension for the given program type.
 * `helper_function_address`: Array of addresses (64-bit unsigned integer) for the helper functions. The addresses must be arranged in the array in the *same order* as the array of helper function prototypes denoted by the `helper_prototype` field in `ebpf_program_info_t` struct.  For the correct execution of eBPF programs, the helper function addresses cannot change while a loaded eBPF program is executing.
 
+There are two sets if helper function addresses that the extension can return. The first are helper functions that are functions that are only callable during execution of an eBPF program that matches this program type. The second are helper functions that override the global helper functions provided by the eBPF runtime.
+
+### `ebpf_program_context_create_t` Function
+This optional function is used to build a program type specific context structure that is used when the application call `bpf_prog_test_run_opts`. The application optionally passes in flat buffers representing the data and the context structure to be passed to the eBPF program. The extension then constructs a context structure to be passed to the eBPF program.
+
+### `ebpf_program_context_destroy_t` Function
+This optional function is used to populate the flat buffers representing the data and context structures that are returned to the application when the `bpf_prog_test_run_opts` call completes. In addition, the function frees any resources allocated in the `ebpf_program_context_create_t` call.
```

Comment:
This says it's optional, but is it required if ebpf_program_context_create is non-null?